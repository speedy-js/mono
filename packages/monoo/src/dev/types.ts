/**
 * Module dependencies
 */
import { IMonorepoBuilderOptions } from '@monoo/types';

export interface IDOptions extends IMonorepoBuilderOptions {}
