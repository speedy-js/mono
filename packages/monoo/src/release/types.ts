/**
 * Module dependencies
 */
import type { RequestInit } from "node-fetch";
import type urlJoin from "url-join";
import { IPackageJSON } from "@monoo/types"

/**
 * Shared namespace
 */
export namespace SharedNS {
  export interface IOptions {
    cwd?: string;
  }
}

/**
 * Release namespace
 */
export namespace ReleaseNS {
  export interface IModifyReleaseArgumentsContext {
    arguments: string[];
    version: string;
    tag: string;
  }

  export interface IReadyContext {
    cwd: string;
  }

  export interface IReleasedContext {
    cwd: string;
    version: string;
    tag: string;
  }

  export interface IPromptAnswers {
    bump: string;
    customVersion: string;
    npmTag: string;
    customNpmTag: string;
  }

  export interface IOptions extends SharedNS.IOptions {
    /**
     * Generate changelog or not.
     */
    changelog?: boolean;
    /**
     * Run in band or not.
     */
    runInBand?: boolean;
    /**
     * Ignore scripts, it will ignore the npm life cycle hooks (e.g. prepublishOnly).
     */
    ignoreScripts?: boolean;
    /**
     * Execute custom build script before release.
     *
     * It is recommended to run topological build before actual publishing process,
     * and then ignore the build script (with `ignoreScripts`) of each package during
     * actual publishing release.
     */
    build?: boolean | string;
    /**
     * Preview execution.
     */
    dryRun?: boolean;
    /**
     * @deprecated
     */
    ready?: (ctx: IReadyContext) => Promise<unknown>;
    /**
     * @deprecated
     */
    modifyReleaseArguments?: (ctx: IModifyReleaseArgumentsContext) => string[];
    /**
     * @deprecated
     */
    released?: (ctx: IReleasedContext) => Promise<unknown>;
  }
}

/**
 * Run namespace
 */
export namespace RunNS {
  export interface IOptions extends SharedNS.IOptions {
    commands: string[];
    runInBand: boolean;
    filter?(v: IPackageJSON): boolean;
    [key: string]: unknown;
  }
}

/**
 * Disttag namespace
 */
export namespace DistTagNS {
  export interface IOptions extends SharedNS.IOptions {
    distVersion: string;
    tag: string;
  }
}

/**
 * Path namespace
 */
export namespace PatchNS {
  export interface IOptions extends SharedNS.IOptions {
    version: string;
    tag: string;
    runInBand: boolean;
    ignoreScripts: boolean;
  }
}

/**
 * Dingding namespace
 */
export namespace DingdingNS {
  export interface IWebhook {
    name: string;
    webhook: string;
    secret: string;
  }

  export interface IOptions extends IWebhook {
    body?: {
      msgtype: "markdown";
      markdown: {
        title: string;
        text: string;
      };
    };
    example?: boolean;
  }

  export interface IWebhookConfig {
    /**
     * Webhooks to post
     */
    webhooks: DingdingNS.IWebhook[];
    /**
     * Test webhook
     */
    testWebhook?: DingdingNS.IWebhook;
    /**
     * Using test webhook or not
     */
    test?: boolean;
  }
}

/**
 * Yuque namespace
 */
export namespace YuqueNS {
  /**
   * Transformer context
   */
  export interface ITransformerContext {
    host: string;
    urlJoin: typeof urlJoin;
    transformerOptions: Record<string, unknown>;
    options: IOptions;
  }

  /**
   * Post data
   */
  export interface IYuquePostData {
    title: string;
    cover: string;
    description: string;
    custom_description: string;
    book: {
      namespace: string;
      user: {
        avatar_url: string;
      };
    };
    slug: string;
  }

  /**
   * Transformer
   */
  export type Transformer = (
    data: IYuquePostData,
    ctx: ITransformerContext
  ) => Record<string, unknown>;

  /**
   * Options
   */
  export interface IOptions
    extends SharedNS.IOptions,
      DingdingNS.IWebhookConfig {
    /**
     * Url of yuque post
     */
    url?: string;
    /**
     * Host of yuque
     * @default 'https://yuque.antfin-inc.com/api/v2'
     */
    host?: string;
    /**
     * Cover to post
     */
    cover?: string;
    /**
     * Endpoint of yuque open api
     */
    endpoint?: string;
    /**
     * Request options
     */
    options?: Partial<RequestInit>;
    /**
     * Transformer preset
     */
    transformerPreset?: "dingding-actionCard";
    /**
     * Transformer options
     */
    transformerOptions?: Record<string, unknown>;
    /**
     * Custom transformer
     */
    transformer?: Transformer;
    /**
     * Log response
     */
    log?: boolean;
  }
}

/**
 * changelog namespace
 */
export namespace ChangelogNS {
  export interface ICommitAuthorOptions {
    /**
     * Attach author to each changelog commit
     *
     * @default false
     */
    attachAuthor?: boolean;
    /**
     * Type of author name
     *
     * @default 'name'
     */
    authorNameType?: "name" | "email";
    /**
     * Get author remote page
     *
     * @param author
     */
    getAuthorPage?(author: ICommitAuthor): string;
  }

  export interface IOptions extends SharedNS.IOptions, ICommitAuthorOptions {
    /**
     * current Version
     *
     * @default version from lerna.json
     */
    version?: string;
    /**
     * Beautify changelog, Remove top-level useless header
     *
     * @see https://github.com/conventional-changelog/conventional-changelog/issues/376
     * @default false
     */
    beautify?: boolean;
    /**
     * Create changelog commit
     *
     * @default false
     */
    commit?: boolean;
    /**
     * Create git push action
     *
     * @default false
     */
    gitPush?: boolean;
  }
  /**
   * Commit author
   */
  export interface ICommitAuthor {
    /**
     * Commit author name
     */
    name: string;
    /**
     * Email name
     */
    emailName: string;
    /**
     * Full email
     */
    email: string;
  }
}

/**
 * PostChangelog namespace
 */
export namespace PostChangelogNS {
  /**
   * View all link
   */
  export interface IViewAll {
    text: string;
    url: string;
  }
  /**
   * Options to infer changelog
   */
  export interface IInferChangelogOptions
    extends SharedNS.IOptions,
      ChangelogNS.ICommitAuthorOptions {
    /**
     * Package name
     */
    packageName: string;
    /**
     * changelog file name
     *
     * @default 'CHANGELOG.md'
     */
    file?: string;
    /**
     * changelog preset, avilable presets: conventional-changelog、afx
     *
     * @default 'conventional-changelog'
     */
    preset?: string;
    /**
     * custom changelog parser
     */
    parser?: (
      changelog: string,
      packageName: string
    ) => {
      title: string;
      content: string;
    };
    /**
     * view all config
     */
    viewAll?: IViewAll;
  }

  export interface IOptions
    extends IInferChangelogOptions,
      DingdingNS.IWebhookConfig {
    /**
     * Custom title, defaults to inferred title from changelog.
     */
    title?: string;
    /**
     * Custom content, defaults to inferred content from changelog.
     */
    content?: string;
  }
}

export type IConfig = ReleaseNS.IOptions & {
  verbose?: boolean;
  postChangelog?: PostChangelogNS.IOptions;
  yuque?: YuqueNS.IOptions;
  dingding?: DingdingNS.IOptions;
  patch?: PatchNS.IOptions;
  distTag?: DistTagNS.IOptions;
  run?: RunNS.IOptions;
};

export function defineConfig(config: IConfig): IConfig {
  return config;
}
