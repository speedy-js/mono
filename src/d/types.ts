/**
 * Module dependencies
 */
import { IMonorepoBuilderOptions } from '../types';

export interface IDOptions extends IMonorepoBuilderOptions {}
