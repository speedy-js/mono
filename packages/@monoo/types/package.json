{
  "name": "@monoo/types",
  "description": "Types for monoo",
  "version": "0.4.3",
  "main": "lib/index",
  "types": "lib/index",
  "files": [
    "lib"
  ],
  "scripts": {
    "dev": "pnpm run build -w",
    "clean": "rimraf ./lib *.tsbuildinfo",
    "build": "tsc -b tsconfig.build.json",
    "prepublishOnly": "pnpm run build"
  },
  "dependencies": {
    "@types/node": "16"
  },
  "devDependencies": {
    "rimraf": "4.1.0"
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org",
    "access": "public"
  }
}
