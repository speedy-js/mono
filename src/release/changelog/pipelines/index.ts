export * from './beautify';
export * from './attach-author';
export * from './shared';
export * from './normalize-commit-url';
