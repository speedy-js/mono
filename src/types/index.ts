export * from './use-config';
export * from './server';
export * from './monorepo';
export * from './monorepo-builder';
