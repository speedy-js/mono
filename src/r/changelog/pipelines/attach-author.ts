/**
 * Module dependencies
 */
import { ChangelogNS } from '../../types';
import { TChangelogProcessor } from './shared';
import { getCommitHashAuthorMap } from '../helpers';
import { COMMIT_RE } from '../regexp';

export interface IDisplayAuthorOptions {
  getAuthorPage?(author: ChangelogNS.ICommitAuthor): string;
  displayType?: 'name' | 'email';
}

export const ChangelogProcessor4AttachAuthor: TChangelogProcessor<IDisplayAuthorOptions> =
  (cwd, input, options = {}) => {
    const displayType = options.displayType ?? 'name';
    const getAuthorPage =
      options.getAuthorPage ??
      ((author: ChangelogNS.ICommitAuthor) => {
        return `https://github.com/${author.emailName}`;
      });

    const hashAuthorMap = getCommitHashAuthorMap(cwd);
    const ret = input.replace(
      COMMIT_RE,
      (
        m,
        s1 /* hash */,
        s2 /* url */,
        s3 /* white space */,
        s4 /* author part */,
        s5 /* author text */,
      ): string => {
        const author = hashAuthorMap[s1];
        /**
         * s4 exists, do not need generate author
         */
        if (!author || s4) {
          return m;
        }
        const dispalyName =
          displayType === 'name' ? author.name : author.emailName;
        const authorUrl = getAuthorPage(author);
        return `([${s1}](${s2})) [@${dispalyName}](${authorUrl})`;
      },
    );

    return ret;
  };
