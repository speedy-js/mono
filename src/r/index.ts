/**
 * Module dependencies
 */
import chalk from 'chalk';
import execa from 'execa';
import semver, { ReleaseType } from 'semver';
import inquirer from 'inquirer';
import { patch } from './patch';
import {
  logger,
  resolveLernaConfig,
  ensureLernaConfig,
  cleanLernaConfig,
} from './shared';
import { ReleaseNS } from './types';
import { changelog } from './changelog';

export { patch, changelog };

/**
 * Select version and tag to be released.
 */
async function selectVersionAndTag(currentVersion: string) {
  const customItem = { name: 'Custom', value: 'custom' };
  const bumps: ReleaseType[] = [
    'patch',
    'minor',
    'major',
    'prerelease',
    'premajor',
  ];
  const versions = bumps.reduce<Record<string, string>>((memo, bump) => {
    memo[bump] = semver.inc(currentVersion, bump);
    return memo;
  }, {});

  const bumpChoices = bumps.map((b) => ({
    name: `${b} (${versions[b]})`,
    value: b,
  }));

  function getVersion(answers: ReleaseNS.IPromptAnswers) {
    return answers.customVersion || versions[answers.bump];
  }

  function getNpmTags(version: string) {
    if (isPreRelease(version)) {
      return ['next', 'latest', 'beta', customItem];
    }
    return ['latest', 'next', 'beta', customItem];
  }

  function isPreRelease(version: string) {
    return Boolean(semver.prerelease(version));
  }

  const { bump, customVersion, npmTag, customNpmTag } =
    await inquirer.prompt<ReleaseNS.IPromptAnswers>([
      {
        name: 'bump',
        message: 'Select release type:',
        type: 'list',
        choices: [...bumpChoices, customItem],
      },
      {
        name: 'customVersion',
        message: 'Input version:',
        type: 'input',
        when: (answers) => answers.bump === 'custom',
      },
      {
        name: 'npmTag',
        message: 'Input npm tag:',
        type: 'list',
        default: (answers: ReleaseNS.IPromptAnswers) =>
          getNpmTags(getVersion(answers))[0],
        choices: (answers) => getNpmTags(getVersion(answers)),
      },
      {
        name: 'customNpmTag',
        message: 'Input customized npm tag:',
        type: 'input',
        when: (answers) => answers.npmTag === 'custom',
      },
    ]);

  const version = customVersion || versions[bump];
  const tag = customNpmTag || npmTag;
  return {
    tag,
    version,
  };
}

export async function release(options: ReleaseNS.IOptions) {
  const cwd = options.cwd ?? process.cwd();

  const config = resolveLernaConfig(cwd);

  if (!config?.path) {
    throw new Error('[MONO] "mono.json" or "lerna.json" doesn\'t exist');
  }

  const lernaConfig: ReleaseNS.ILernaConfig = config.data;

  if (lernaConfig.version === 'independent') {
    throw new Error(
      '[MONO] "release" cannot be executed under "independent" mode',
    );
  }

  if (typeof options.ready === 'function') {
    await options.ready({ cwd });
  }

  logger.info(`${chalk.gray('Current version: ')}${lernaConfig.version}`);

  const { version, tag } = await selectVersionAndTag(lernaConfig.version);

  const { yes } = await inquirer.prompt([
    {
      name: 'yes',
      message: `Confirm releasing ${version} (${tag})?`,
      type: 'list',
      choices: ['N', 'Y'],
    },
  ]);

  if (yes === 'N') {
    console.log('[MONO] cancelled.');
    return;
  }

  /**
   * Set this env for subsequent build process.
   */
  process.env.MONO_RELEASE_VERSION = version;

  /**
   * Execute custom build script before release.
   */
  if (options.build) {
    const buildScript =
      typeof options.build === 'string' ? options.build : 'npm run build';
    const [command, ...args] = buildScript.split(' ');
    await execa(command, args, {
      shell: true,
      cwd: options.cwd,
      stdio: 'inherit',
      env: process.env,
    });
  }

  let releaseArguments: string[] = [
    'publish',
    version,
    '--exact',
    '--force-publish',
    '--dist-tag',
    tag,
  ];

  if (typeof options.modifyReleaseArguments === 'function') {
    releaseArguments = options.modifyReleaseArguments({
      arguments: releaseArguments,
      version,
      tag,
    });
  }

  if (options.ignoreScripts) {
    releaseArguments.push('--ignore-scripts');
  }

  ensureLernaConfig(cwd);

  try {
    await execa(
      require.resolve('lerna/cli.js', {
        paths: [cwd, __dirname],
      }),
      releaseArguments,
      {
        stdio: 'inherit',
      },
    );
  } catch (e) {
    console.log(e);

    await patch({
      cwd,
      version,
      tag,
      runInBand: options.runInBand,
      ignoreScripts: options.ignoreScripts,
    });
  }

  if (typeof options.released === 'function') {
    await options.released({ cwd, version, tag });
  }

  cleanLernaConfig(cwd);

  if (options.changelog) {
    await changelog({
      cwd,
      beautify: true,
      commit: true,
      gitPush: true,
      attachAuthor: true,
      authorNameType: 'email',
    });
  }
}
