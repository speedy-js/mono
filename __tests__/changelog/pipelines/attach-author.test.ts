/**
 * Module dependencies
 */
import { ChangelogProcessor4AttachAuthor } from '../../../src/r/changelog/pipelines/attach-author';
import { ChangelogNS } from '../../../src/r/types';

jest.mock(require.resolve('../../../src/r/changelog/helpers'), () => {
  const mockedAuthor: ChangelogNS.ICommitAuthor = {
    name: '<name>',
    email: '<email>',
    emailName: '<emailName>',
  };

  const mockedAuthor2: ChangelogNS.ICommitAuthor = {
    name: '<name2>',
    email: '<email2>',
    emailName: '<emailName2>',
  };

  return {
    getCommitHashAuthorMap: () => {
      return {
        aaaaaaa: mockedAuthor,
        bbbbbbb: mockedAuthor,
        ccccccc: mockedAuthor2,
        ddddddd: mockedAuthor2,
      };
    },
  };
});

describe('ChangelogProcessor4AttachAuthor', () => {
  describe('update changelog without author', () => {
    const input = `
## [0.2.4](https://github.com/speedy/mono/compare/v0.2.3...v0.2.4) (2021-10-23)


### Bug Fixes

* \`--ignore-scripts\` description ([aaaaaaa](https://github.com/speedy/mono/commits/663a3963b63706d273068895f4dd79d6c70472c2))



## [0.2.3](https://github.com/speedy/mono/compare/v0.2.2...v0.2.3) (2021-10-20)


### Bug Fixes

* The "path" argument must be of type string. Received undefined ([bbbbbbb](https://github.com/speedy/mono/commits/1a377bb70193547e2cf9faaa288fbf5fcdc4b66f))
  `;

    it('default', () => {
      expect(
        ChangelogProcessor4AttachAuthor('<cwd>', input, {}),
      ).toMatchSnapshot();
    });

    it('options - displayType', () => {
      expect(
        ChangelogProcessor4AttachAuthor('<cwd>', input, {
          displayType: 'email',
        }),
      ).toMatchSnapshot();
    });

    it('options - getAuthorPage', () => {
      expect(
        ChangelogProcessor4AttachAuthor('<cwd>', input, {
          getAuthorPage(author) {
            return `/page/to/author/${author.emailName}`;
          },
        }),
      ).toMatchSnapshot();
    });
  });

  describe('update changelog with author', () => {
    const input = `
## [0.2.4](https://github.com/speedy/mono/compare/v0.2.3...v0.2.4) (2021-10-23)


### Bug Fixes

* \`--ignore-scripts\` description ([aaaaaaa](https://github.com/speedy/mono/commits/663a3963b63706d273068895f4dd79d6c70472c2))



## [0.2.3](https://github.com/speedy/mono/compare/v0.2.2...v0.2.3) (2021-10-20)


### Bug Fixes

* The "path" argument must be of type string. Received undefined ([bbbbbbb](https://github.com/speedy/mono/commits/1a377bb70193547e2cf9faaa288fbf5fcdc4b66f)) [@<name>](https://github.com/<emailName>)
  `;

    it('default', () => {
      expect(
        ChangelogProcessor4AttachAuthor('<cwd>', input, {}),
      ).toMatchSnapshot();
    });

    it('options - displayType', () => {
      expect(
        ChangelogProcessor4AttachAuthor('<cwd>', input, {
          displayType: 'email',
        }),
      ).toMatchSnapshot();
    });

    it('options - getAuthorPage', () => {
      expect(
        ChangelogProcessor4AttachAuthor('<cwd>', input, {
          getAuthorPage(author) {
            return `/page/to/author/${author.emailName}`;
          },
        }),
      ).toMatchSnapshot();
    });
  });
});
