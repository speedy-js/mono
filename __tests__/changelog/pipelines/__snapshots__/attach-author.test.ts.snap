// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangelogProcessor4AttachAuthor update changelog with author default 1`] = `
"
## [0.2.4](https://github.com/speedy/mono/compare/v0.2.3...v0.2.4) (2021-10-23)


### Bug Fixes

* \`--ignore-scripts\` description ([aaaaaaa](https://github.com/speedy/mono/commits/663a3963b63706d273068895f4dd79d6c70472c2)) [@<name>](https://github.com/<emailName>)



## [0.2.3](https://github.com/speedy/mono/compare/v0.2.2...v0.2.3) (2021-10-20)


### Bug Fixes

* The \\"path\\" argument must be of type string. Received undefined ([bbbbbbb](https://github.com/speedy/mono/commits/1a377bb70193547e2cf9faaa288fbf5fcdc4b66f)) [@<name>](https://github.com/<emailName>)
  "
`;

exports[`ChangelogProcessor4AttachAuthor update changelog with author options - displayType 1`] = `
"
## [0.2.4](https://github.com/speedy/mono/compare/v0.2.3...v0.2.4) (2021-10-23)


### Bug Fixes

* \`--ignore-scripts\` description ([aaaaaaa](https://github.com/speedy/mono/commits/663a3963b63706d273068895f4dd79d6c70472c2)) [@<emailName>](https://github.com/<emailName>)



## [0.2.3](https://github.com/speedy/mono/compare/v0.2.2...v0.2.3) (2021-10-20)


### Bug Fixes

* The \\"path\\" argument must be of type string. Received undefined ([bbbbbbb](https://github.com/speedy/mono/commits/1a377bb70193547e2cf9faaa288fbf5fcdc4b66f)) [@<name>](https://github.com/<emailName>)
  "
`;

exports[`ChangelogProcessor4AttachAuthor update changelog with author options - getAuthorPage 1`] = `
"
## [0.2.4](https://github.com/speedy/mono/compare/v0.2.3...v0.2.4) (2021-10-23)


### Bug Fixes

* \`--ignore-scripts\` description ([aaaaaaa](https://github.com/speedy/mono/commits/663a3963b63706d273068895f4dd79d6c70472c2)) [@<name>](/page/to/author/<emailName>)



## [0.2.3](https://github.com/speedy/mono/compare/v0.2.2...v0.2.3) (2021-10-20)


### Bug Fixes

* The \\"path\\" argument must be of type string. Received undefined ([bbbbbbb](https://github.com/speedy/mono/commits/1a377bb70193547e2cf9faaa288fbf5fcdc4b66f)) [@<name>](https://github.com/<emailName>)
  "
`;

exports[`ChangelogProcessor4AttachAuthor update changelog without author default 1`] = `
"
## [0.2.4](https://github.com/speedy/mono/compare/v0.2.3...v0.2.4) (2021-10-23)


### Bug Fixes

* \`--ignore-scripts\` description ([aaaaaaa](https://github.com/speedy/mono/commits/663a3963b63706d273068895f4dd79d6c70472c2)) [@<name>](https://github.com/<emailName>)



## [0.2.3](https://github.com/speedy/mono/compare/v0.2.2...v0.2.3) (2021-10-20)


### Bug Fixes

* The \\"path\\" argument must be of type string. Received undefined ([bbbbbbb](https://github.com/speedy/mono/commits/1a377bb70193547e2cf9faaa288fbf5fcdc4b66f)) [@<name>](https://github.com/<emailName>)
  "
`;

exports[`ChangelogProcessor4AttachAuthor update changelog without author options - displayType 1`] = `
"
## [0.2.4](https://github.com/speedy/mono/compare/v0.2.3...v0.2.4) (2021-10-23)


### Bug Fixes

* \`--ignore-scripts\` description ([aaaaaaa](https://github.com/speedy/mono/commits/663a3963b63706d273068895f4dd79d6c70472c2)) [@<emailName>](https://github.com/<emailName>)



## [0.2.3](https://github.com/speedy/mono/compare/v0.2.2...v0.2.3) (2021-10-20)


### Bug Fixes

* The \\"path\\" argument must be of type string. Received undefined ([bbbbbbb](https://github.com/speedy/mono/commits/1a377bb70193547e2cf9faaa288fbf5fcdc4b66f)) [@<emailName>](https://github.com/<emailName>)
  "
`;

exports[`ChangelogProcessor4AttachAuthor update changelog without author options - getAuthorPage 1`] = `
"
## [0.2.4](https://github.com/speedy/mono/compare/v0.2.3...v0.2.4) (2021-10-23)


### Bug Fixes

* \`--ignore-scripts\` description ([aaaaaaa](https://github.com/speedy/mono/commits/663a3963b63706d273068895f4dd79d6c70472c2)) [@<name>](/page/to/author/<emailName>)



## [0.2.3](https://github.com/speedy/mono/compare/v0.2.2...v0.2.3) (2021-10-20)


### Bug Fixes

* The \\"path\\" argument must be of type string. Received undefined ([bbbbbbb](https://github.com/speedy/mono/commits/1a377bb70193547e2cf9faaa288fbf5fcdc4b66f)) [@<name>](/page/to/author/<emailName>)
  "
`;
